Update Readme (#30)
brown-ccv/volume-data-converter: README.md

# Volume-data-converter

Project to convert tiff image sequences ( 3D volume ) to a 2D png downscaled texture atlas. This application can be used to port volumes to the [aframe volume viewer](https://github.com/brown-ccv/react-volume-viewer).

> ## Suppoort to Rhode Island Discovery and Dada Center Volume Viewer

Reviewer: ```suggestion
## Suppoort to Rhode Island Discovery and Data Center Volume Viewer
```

## Prerequisites

- Python > 3.8 or < 3.11
- Pip installed in your environment
- (optional) [Python Poetry](https://github.com/python-poetry/poetry)

## Installation

1. Clone the repo, go to the project's directory.
2. Create a virtual environment using the command  Mac/Unix: `python3 -m venv ./` Windows: `python3 -m venv .\`
3. Activate the virtual environment Mac/Unix: `source venv/bin/activate` Windows: `.\venv\Scripts\activate`
4. Install dependencies:

### Using Python Poetry

- Run `poetry install`  

### Using Pip package managment

- Run `pip install -r requirements.txt`  

The command will install the dependencies for the project.

## Running the available commands

Activate the virtual environment to run the following commands (If you are using poetry you can execute the command by pre-appending the command `poetry run `):

### Converting tiff stack of images to png texture map

To verify the command runs correctly, run the command:

`convert2png --help`

> You should see the available paramaters that can be used.

Reviewer: ```suggestion
You should see the available parameters that can be used.
```

        "time>=": "2019-12-30T00:00:00Z",
        "time<=": "2019-12-31T00:00:00Z",
        "time_step": 1,
        "eta_rho>=": 0,
        "eta_rho<=": 1099,
        "eta_rho_step": 1,
        "xi_rho>=": 0,
        "xi_rho<=": 999,
        "xi_rho_step": 1
  }
}
```

Modify the imporant attributes such as `server ULR`, `dataset_id` and `contrainst` according to the data you want to visualize.

There is a default `erddap_configuration.json` file located in the `configuration` folder.

This is an example on how to run the command:

`erddap_query path_to_result_folder path_to_configuration_file.json`

### Converting data from NC to .raw or .png

The command `osom-converter` has only one parameter:

- `parameters_file_path`: Path to a `.json` file that looks like this:

```json
{
  "osom_grid_file": "path_to_grid_file/osom_grid4_mindep_smlp_mod7-netcdf4.nc",
  "osom_data_file": "path_to_nc_datafile/model_data_57db_4a85_81d9.nc",
  "output_folder": "path_to_output_folder",
  "data_descriptor" : "WaterTempSurface",
  "time_frames": [0],
  "layer" : "surface",
  "to_texture_atlas": true
}
```

- There is an example grid file located in the `resources\grid-data` folder

> - The `osom_data_file` is the resulting osom-nc file fomr the `erddap_query` command.

Reviewer: ```suggestion
- The `osom_data_file` is the resulting osom-nc file from the `erddap_query` command.
```

- `data_descriptor` is one of the variables in the erddap data base.

> - When the osom-nc files respresent multiple time frames ( i.e data representing different hours in a day ), the data is store in an 1D-array. `time_frames` argument specify the time frame you want to convert into volume viewer data. By default, it will convert all the available time frames.

Reviewer: ```suggestion
- When the osom-nc files respresent multiple time frames ( i.e data representing different hours in a day ), the data is stored in an 1D-array. The `time_frames` argument specify the time frame you want to convert into volume viewer data. By default, it will convert all the available time frames.
```

> - Some datasets are multilayered. The `layer` argument let you specify if the dataset has only one  (`surface`, `bottom`) or multiple (`all`) layers

Reviewer: ```suggestion
- Some datasets are multi-layered. The `layer` argument lets you specify if the dataset has only one layer (`surface`, `bottom`) or multiple layers (`all`).
```

> - `to_texture_atlas` set to True will convert the nc files into a 2D png texture atlas that can be readed by the [aframe volume viewer](https://github.com/brown-ccv/react-volume-viewer).

Reviewer: ```suggestion
- `to_texture_atlas` set to True will convert the nc files into a 2D png texture atlas that can be read by the [aframe volume viewer](https://github.com/brown-ccv/react-volume-viewer).
```
